java/src/jmri/swing/JmriJTablePersistenceManager.java

```diff
@@ -600,6 +600,7 @@ public void columnAdded(TableColumnModelEvent e) {
 
         @Override
         public void columnRemoved(TableColumnModelEvent e) {
+            this.manager.clearState(this.table); // deletes column data from xml file
```

Comment:
```clearState(JTable)``` does not delete column data from a file, it merely causes the manager to forget the state of that table.